### PR TITLE
MMDP-488 Manage Target Audience Dropdown options.

### PR DESCRIPTION
**Pull request by Musacoli**
**What does this PR do?**

Implements the ability for an `administrator` to `manage` options in the `Target Audience` Dropdown.

**Description of Task to be completed?**
- [x] implement `create` Target Audience endpoint
- [x] implement `update` Target Audience endpoint
- [x] implement `get` Target Audience endpoint
- [x] implement `delete` Target Audience endpoint
- [x] consume `create` Target Audience endpoint
- [x] consume `update` Target Audience endpoint
- [x] consume `get` Target Audience endpoint
- [x] consume `delete` Target Audience endpoint

**How should this be manually tested?**

- Pull and checkout to`ft-target-audience-dropdown-MMDP-488` branch
- run `yarn dev-server` 
- route to `localhost:1234/dropdowns/target-audience`
- Fill in the form with all required fields and click the save button

**Any background context you want to provide?**

n/a

**What are the relevant Jira issues?**

[MMDP-488](https://viisaus.atlassian.net/browse/MMDP-488)

**Screenshots (if appropriate)**
**With 1 option**

**With more added options**

**Questions:**
N/A

### CODE REVIEW

**Comment by bozicschucky** [comment]
Hey @Musacoli, great work but the toaster for when a dropdown is deleted successfully shows a warning.
Kindly fix that. Thanks
Screenshot 2019-04-01 at 11 38 44

**Comment by bozicschucky** [comment]
Hey @Musacoli updating a dropdown doesn't work and shows a validation error which is not descriptive enough, do you mind fixing that.

**Comment by Ruiru11** [comment]
@Musacoli when I add more than one dropdowns it just says target audience option successfully added. ideally, it should say the number of options added, kindly fix that so that we maintain consistency across all dropdowns 
image
and also from a discussion I had with Francis description fields should be optional (you could look into that as yours are not)
image

File: client/store/sagas/dropdowns/targetAudience.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      type: FETCHING_TARGET_AUDIENCES,
+    });
+    toastr.success('Target Audience delete successful');
+  } catch (err) {
+    yield put({
+      type: 'DELETE_TARGET_AUDIENCE_FAILURE',
+      payload: { error: err.response.data.message },
+    });
+    const message = err.response.data.message;
+    toastr.warning(message);
```
Comment (shemogumbe) [inline client/store/sagas/dropdowns/targetAudience.js]: This ought not to be a warning but an error

**Pull request merged**